src/data_readers/cv_utils.cpp

```diff
@@ -102,9 +102,13 @@ cv::Mat cv_utils::lbann_imread(const std::string& img_file_path, int flags, std:
   const cv::Mat inbuf(1, buf.size(), InputBuf_T::T(1), buf.data());
 
   // decode the image data in the memory buffer
-  cv::Mat image = cv::imdecode(inbuf, cv::IMREAD_ANYCOLOR | cv::IMREAD_ANYDEPTH);
-
-  return image;
+  if(cv_buf != nullptr) {
```

Comment:
It might be the OpenCV's bug if it does not work without this conditional.
According to https://github.com/opencv/opencv/blob/master/modules/imgcodecs/src/loadsave.cpp +856, the returned Mat is the same as *cv_buf if it is not null.